homeassistant/components/kodi/media_player.py

```diff
@@ -118,118 +116,110 @@
-        port = discovery_info.get("port")
-        tcp_port = DEFAULT_TCP_PORT
-        encryption = DEFAULT_PROXY_SSL
-        websocket = DEFAULT_ENABLE_WEBSOCKET
-        properties = discovery_info.get("properties")
-        if properties is not None:
-            unique_id = properties.get("uuid", None)
-
-    # Only add a device once, so discovered devices do not override manual
-    # config.
-    ip_addr = socket.gethostbyname(host)
-    if ip_addr in hass.data[DOMAIN]:
+    if discovery_info:
+        # Now handled by zeroconf in the config flow
+        return
+
+    host = config[CONF_HOST]
+    if find_matching_config_entries_for_host(hass, host):
         return
 
-    # If we got an unique id, check that it does not exist already.
-    # This is necessary as netdisco does not deterministally return the same
-    # advertisement when the service is offered over multiple IP addresses.
-    if unique_id is not None:
-        for device in hass.data[DOMAIN].values():
-            if device.unique_id == unique_id:
-                return
-
-    entity = KodiDevice(
-        hass,
-        name=name,
-        host=host,
-        port=port,
-        tcp_port=tcp_port,
-        encryption=encryption,
-        username=config.get(CONF_USERNAME),
-        password=config.get(CONF_PASSWORD),
-        turn_on_action=config.get(CONF_TURN_ON_ACTION),
-        turn_off_action=config.get(CONF_TURN_OFF_ACTION),
-        timeout=config.get(CONF_TIMEOUT),
-        websocket=websocket,
-        unique_id=unique_id,
+    websocket = config.get(CONF_ENABLE_WEBSOCKET)
+    ws_port = config.get(CONF_TCP_PORT) if websocket else None
+
+    entry_data = {
+        CONF_NAME: config.get(CONF_NAME, host),
+        CONF_HOST: host,
+        CONF_PORT: config.get(CONF_PORT),
+        CONF_WS_PORT: ws_port,
+        CONF_USERNAME: config.get(CONF_USERNAME),
+        CONF_PASSWORD: config.get(CONF_PASSWORD),
+        CONF_SSL: config.get(CONF_PROXY_SSL),
+        CONF_TIMEOUT: config.get(CONF_TIMEOUT),
+    }
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=entry_data
+        )
+    )
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Kodi media player platform."""
+    platform = entity_platform.current_platform.get()
+    platform.async_register_entity_service(
+        SERVICE_ADD_MEDIA, KODI_ADD_MEDIA_SCHEMA, "async_add_media_to_playlist"
+    )
+    platform.async_register_entity_service(
+        SERVICE_CALL_METHOD, KODI_CALL_METHOD_SCHEMA, "async_call_method"
     )
 
-    hass.data[DOMAIN][ip_addr] = entity
-    async_add_entities([entity], update_before_add=True)
+    data = hass.data[DOMAIN][config_entry.entry_id]
+    connection = data[DATA_CONNECTION]
+    version = data[DATA_VERSION]
+    kodi = data[DATA_KODI]
+    name = config_entry.data[CONF_NAME]
+    uid = config_entry.data.get(CONF_ID)
+
```

Comment:
when there is no unique_id set, you should use the config entry id so that it can still be edited from UI

Reply:
Cool, I thought this was just for devices that don't have unique ids at all, didn't know it was also a best practice for cases where the devices can have a unique id, but a specific instance doesn't.